static/app/views/insights/common/views/spans/selectors/eapSelector.tsx

```diff
@@ -0,0 +1,88 @@
+import {
+  CompactSelect,
+  type SelectOption,
+  type SelectProps,
+} from 'sentry/components/compactSelect';
+import {useLocation} from 'sentry/utils/useLocation';
+import {useNavigate} from 'sentry/utils/useNavigate';
+import useOrganization from 'sentry/utils/useOrganization';
+
+type EapOption = 'useEap' | 'noEap' | 'useEapRpc';
+
+/**
+ * This selector allows us to with and without EAP enabled
+ * @returns
+ */
+export function EapSelector() {
+  const navigate = useNavigate();
+  const location = useLocation();
+  const {features} = useOrganization();
+  const {useEap, useRpc} = useEapOptions();
+  const hasEAPFeature = features.includes('insights-use-eap-wip');
+
+  type Options = SelectProps<EapOption>['options'];
+
+  const options: Options = [
+    {
+      value: 'noEap',
+      label: "Don't use EAP",
+    },
+    {
+      value: 'useEap',
+      label: 'Use EAP',
+    },
+    {
+      value: 'useEapRpc',
+      label: 'Use RPC EAP',
+    },
+  ];
+
+  let value: EapOption = 'useEap';
+  if (!useEap) {
+    value = 'noEap';
+  } else if (useRpc) {
+    value = 'useEapRpc';
+  }
+
+  if (hasEAPFeature) {
+    return (
+      <CompactSelect
+        options={options}
+        value={value}
+        onChange={(selectedOption: SelectOption<EapOption>) =>
+          navigate({
+            ...location,
+            query: {
+              ...location.query,
+              eap_option: selectedOption.value,
+            },
+          })
+        }
+      />
+    );
+  }
```

Comment:
You can avoid this by returning `undefined` as soon as the feature is available at the top of the component

Reply:
Also, isn't this redundant? `useEapOptions` already checks the feature flag